### ISSUE TITLE
PEP661 sentinels in instance attributes in strict mode result in unknown types

### ISSUE THREAD

**Issue by inducer** [microsoft/pyright#11467]
For this snippet ([playground](https://pyright-play.net/?strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBA%2BjwFcAuCQFM5QJYQA5hCSgEMA7UsJYpLMUgZwChHRIoATa4gYwBtj69NPWx4CRTlT4CmLcNCQBPXFlIBzGGgAeSNA1oNR%2BQlADKemqTS8oUAMRRcikFjUALJAC5saiugDa6MZIALJYgqpqIWDsCLxopmCo3GgAuswAggCqACoA8lAAvGYWqtYAFABE2fmVAJTMjAACkjz8gozSglAZICDEip6Mtrb0bsS4aN41ecwjHGjAUGpoSGMTaOVCvMB1Q-PzuC6kSFvWwAB065MNQA)):
```python
from __future__ import annotations

from dataclasses import dataclass

from typing_extensions import Sentinel  # pyright: ignore[reportMissingModuleSource]

AUTO = Sentinel("AUTO")


@dataclass
class Array:
    shape: AUTO

    def getshape(self):
        print(self.shape)
```
pyright, when configured with
```
[tool.pyright]
strict = ["*.py"]
enableExperimentalFeatures = true
```
says:
```
/home/andreas/tmp/sent2.py
  /home/andreas/tmp/sent2.py:15:15 - warning: Type of "shape" is unknown (reportUnknownMemberType)
  /home/andreas/tmp/sent2.py:15:15 - warning: Argument type is unknown
    Argument corresponds to parameter "values" in function "print" (reportUnknownArgumentType)
0 errors, 2 warnings, 0 notes
```
Other uses of the same type (function argument, global) do not lead to this warning, leading me to suspect there's a bug here.

Version: pyright 1.1.409 (CLI)
